### PR TITLE
Render interface FilePath usage cleanup

### PR DESCRIPTION
**Pull request by bernardkwok**
Changes to use FilePath vs straight strings for file paths.

### CODE REVIEW

File: source/MaterialXRender/Handlers/HwLightHandler.h [hunk 1]
```diff
@@ -57,30 +60,31 @@ class HwLightHandler
...
-    const string& getLightEnvIrradiancePath() const
+    /// Get path to an irradiance IBL image
+    const FilePath& getLightEnvIrradiancePath() const
     {
         return _lightEnvIrradiancePath;
     }
 
-    /// Set path to radiance IBL image
-    void setLightEnvRadiancePath(const string& path)
+    /// Set path to an radiance IBL image
```
Comment (feldstj) [inline source/MaterialXRender/Handlers/HwLightHandler.h]: a not an

**Comment by feldstj** [comment]
Seem to be a lot of places where you converted "string& fileName" to "FilePath& fileName" should this be "FilePath& filePath" instead?

**Comment by niklasharrysson** [comment]
@feldstj @bernardkwok Maybe we could use a convention like this:
```
const FilePath& file - when it points to a specific file
const FilePath& path - when it points to a directory and no specific file
```
 to keep this consistent and clear on what the intended use is.

**Comment by feldstj** [comment]
@niklasharrysson @bernardkwok  Shouldn't it really be filePath and dirPath then? Because I can easily see someone using a FilePath to open an actual file which would make it strange if the filePath was called file.

File: source/MaterialXRenderOsl/OslValidator.cpp [hunk 2]
```diff
@@ -51,14 +51,14 @@ void OslValidator::initialize()
     }
 }
 
-void OslValidator::renderOSL(const string& outputPath, const string& shaderName, const string& outputName)
+void OslValidator::renderOSL(const FilePath& filePath, const string& shaderName, const string& outputName)
```
Comment (niklasharrysson) [inline source/MaterialXRenderOsl/OslValidator.cpp]: Here I guess it's not a file but a directory?

File: source/MaterialXRenderOsl/OslValidator.cpp [hunk 3]
```diff
@@ -177,15 +177,15 @@ void OslValidator::renderOSL(const string& outputPath, const string& shaderName,
     }
 }
 
-void OslValidator::shadeOSL(const string& outputPath, const string& shaderName, const string& outputName)
+void OslValidator::shadeOSL(const FilePath& filePath, const string& shaderName, const string& outputName)
```
Comment (niklasharrysson) [inline source/MaterialXRenderOsl/OslValidator.cpp]: Also a directory?

File: source/MaterialXRenderOsl/OslValidator.h [hunk 4]
```diff
@@ -203,17 +206,17 @@ class OslValidator : public ShaderValidator
   protected:
     ///
     /// Shade using OSO input file. Will throw an exception if an error occurs.
-    /// @param outputPath Path to input .oso file.
+    /// @param filePath Path to input .oso file.
```
Comment (niklasharrysson) [inline source/MaterialXRenderOsl/OslValidator.h]: Somewhat misleading docs (I guess search and replace error). I think this should be an output directory, not a path to an input file?

Comment (niklasharrysson) [reply source/MaterialXRenderOsl/OslValidator.h]: Hmm.. or is it an input path folder. This is slightly confusing :)

**Pull request merged**